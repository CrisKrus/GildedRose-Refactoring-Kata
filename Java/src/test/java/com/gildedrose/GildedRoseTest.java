package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void updateItemQuality() {
        CombinationApprovals.verifyAllCombinations(
            (name, sellIn, quality) -> updateQuality(name, sellIn, quality).toString().trim(),
            new String[]{"foo", "Aged Brie", "Sulfuras, Hand of Ragnaros", "Backstage passes to a TAFKAL80ETC concert"}, // name
            new Integer[]{0, 5, 10, 100}, // sellIn
            new Integer[]{0, 5, 10, 50} // quality
        );
    }

    private static Item updateQuality(String name, int sellIn, int quality) {
        Item item = new Item(name, sellIn, quality);
        GildedRose app = new GildedRose(new Item[] { item });
        app.updateQuality();
        return item;
    }

}
