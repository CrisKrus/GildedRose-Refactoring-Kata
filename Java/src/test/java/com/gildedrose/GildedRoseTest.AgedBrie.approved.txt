Aged Brie[0] = Aged Brie, -1, 2
Aged Brie[1] = Aged Brie, 4, 1
Aged Brie[2] = Aged Brie, 9, 1
Aged Brie[3] = Aged Brie, -1, 7
Aged Brie[4] = Aged Brie, 4, 6
Aged Brie[5] = Aged Brie, 9, 6
